src/core/main.c

```diff
@@ -146,6 +146,11 @@ static NUMAPolicy arg_numa_policy;
 
 static int parse_configuration(void);
 
+/* The original rlimits we passed in. Note we use different values for the two that indicate
+* whether these fields are initialized! */
+static struct rlimit saved_rlimit_nofile = RLIMIT_MAKE_CONST(0);
+static struct rlimit saved_rlimit_memlock = RLIMIT_MAKE_CONST(RLIM_INFINITY);
```

Comment:
why make this a global static variable? we usually avoid that whereever we can

Reply:
I know you won't like it ;)

Well I thought we could make an exception here since they're used at various places (but their usages still remains inside main.c) and making them static make the code simpler and easier to read IMHO.

Reply:
dunno, i have trouble grokking what this patch does, and i have the suspicion this would be much either to grok without this refactoring...